### PR TITLE
Removing apim

### PR DESCRIPTION
**Pull request by antonioT90**
### List of changes

apim removed from gateway, it will be temporarely used only for mock purpose

### Type of changes

- [ ] Add new resources
- [x] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [x] DEV
- [ ] UAT
- [ ] PROD

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [x] No

### Other information

---

### CODE REVIEW

File: src/core/appgateway.tf [hunk 1]
```diff
@@ -52,13 +52,13 @@ module "app_gw" {
...
+    aks = {
+      protocol     = "Http"
       host         = trim(azurerm_dns_a_record.dns_a_api.fqdn, ".")
-      port         = 443
-      ip_addresses = module.apim.private_ip_addresses
-      probe        = "/status-0123456789abcdef"
-      probe_name   = "probe-apim"
+      port         = 80
+      ip_addresses = [var.reverse_proxy_ip]
+      probe        = "/spid/v1/metadata"
```
Comment (pasqualedevita) [inline src/core/appgateway.tf:60]: probably we need a more agnostic probe
I tried
```
curl -k https://10.1.0.250:6443/readyz
```
but not works with aks

https://kubernetes.io/docs/reference/using-api/health-checks/

File: src/k8s/selc_ingress.tf [hunk 2]
```diff
@@ -21,50 +57,76 @@ resource "kubernetes_ingress" "selc_ingress" {
...
+    name      = "${kubernetes_namespace.selc.metadata[0].name}-pdnd-ingress"
+    namespace = kubernetes_namespace.selc.metadata[0].name
+    annotations = {
+      "kubernetes.io/ingress.class"                    = "nginx"
+      "nginx.ingress.kubernetes.io/rewrite-target"     = "/pdnd-interop-uservice-$1/0.1/$2"
+      "nginx.ingress.kubernetes.io/ssl-redirect"       = "false"
+      "nginx.ingress.kubernetes.io/use-regex"          = "true"
+      "nginx.ingress.kubernetes.io/enable-cors"        = "true"
+      "nginx.ingress.kubernetes.io/cors-allow-headers" = "*"
+      "nginx.ingress.kubernetes.io/cors-allow-headers" = local.cors.headers
```
Comment (pasqualedevita) [inline src/k8s/selc_ingress.tf:97]: typo?

File: src/k8s/selc_ingress.tf [hunk 3]
```diff
@@ -21,50 +57,76 @@ resource "kubernetes_ingress" "selc_ingress" {
...
+
+resource "kubernetes_ingress" "selc_pdnd_ingress" {
+  depends_on = [helm_release.ingress]
+
+  metadata {
+    name      = "${kubernetes_namespace.selc.metadata[0].name}-pdnd-ingress"
+    namespace = kubernetes_namespace.selc.metadata[0].name
+    annotations = {
+      "kubernetes.io/ingress.class"                    = "nginx"
+      "nginx.ingress.kubernetes.io/rewrite-target"     = "/pdnd-interop-uservice-$1/0.1/$2"
```
Comment (pasqualedevita) [inline src/k8s/selc_ingress.tf:92]: maybe better use `0.1` as terraform var

File: src/k8s/selc_ingress.tf (hunk 2 above)
Comment (antonioT90) [reply src/k8s/selc_ingress.tf]: fixed

File: src/k8s/selc_ingress.tf (hunk 3 above)
Comment (antonioT90) [reply src/k8s/selc_ingress.tf:92]: the version 0.1 is statically mapped to the v1 version of the ingress' path
I suggest to let it be a costant and not defined as variable

File: src/core/appgateway.tf (hunk 1 above)
Comment (antonioT90) [reply src/core/appgateway.tf:60]: I was neither able to find a better probe...
at this moment I would let asis

Comment (antonioT90) [reply src/core/appgateway.tf:60]: I think @diegolagospagopa could provide some useful hint :)

Comment (antonioT90) [reply src/core/appgateway.tf]: as shared, we will use the ms-product health check at this moment

**Review by pasqualedevita — APPROVED** [review]

**Pull request merged**